setup.py

```diff
@@ -10,19 +10,7 @@
 CURRENT_VERSION = __version__
 FRAMEWORKS = ["tensorflow", "pytorch", "mxnet", "xgboost"]
 TESTS_PACKAGES = ["pytest", "torchvision", "pandas"]
-INSTALL_REQUIRES = [
-    # aiboto3 implicitly depends on aiobotocore
-    "aioboto3==6.4.1",  # no version deps
-    "aiobotocore==0.11.0",  # pinned to a specific botocore & boto3
-    "aiohttp>=3.6.0,<4.0",  # aiobotocore breaks with 4.0
-    # boto3 explicitly depends on botocore
-    "boto3>=1.10.32",  # Sagemaker requires >= 1.9.213
-    "botocore>=1.13.32",
-    "nest_asyncio",
-    "protobuf>=3.6.0",
-    "numpy",
-    "packaging",
-]
+INSTALL_REQUIRES = ["protobuf>=3.6.0", "numpy", "packaging", "boto3>=1.10.32"]
```

Comment:
can we pin all these versions to a unique version ?

Reply:
Python dependency management works different than other languages, due to how PyPI dependency resolution works. 

As library developers, we should not be pinning to any specific version. The maintainers of environments which include this library, or users of library can pin to specific versions of dependencies they use. If the library itself pins a specific version of a dependency, then some other package which also depends on that same dependency in the same environment can't be installed due to version conflicts. 

Here's a reference https://stackoverflow.com/a/44938662

Reply:
We should not block this PR.  Let's talk about pinning, goals behind it and how can we achieve those goals.